src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -276,7 +276,6 @@ class ReimbursementAccountPage extends React.Component {
                 <ContinueBankAccountSetup
                     reimbursementAccount={this.props.reimbursementAccount}
                     continue={this.continue}
-                    startOver={() => this.setState({shouldHideContinueSetupButton: true})}
```

Comment:
No need to set this state manually, the `reimbursementAccount` will be cleared if we succeed and `shouldHideContinueSetupButton` won't be used.

Reply:
Can you elaborate on this one? After resetting `reimbursementAccount`, `shouldHideContinueSetupButton` won't be recalculated. Or are you relaying on [componentDidUpdate](https://github.com/Expensify/App/blob/3e15034b5e768393e56e7b6bf884bd2e3e61b135/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L99-L114) to redirect you hoping the steps to be different?

Reply:
I think there is a bug now, the `shouldHideContinueSetupButton` remains `false` during the next setup if you don't close/reopen the modal. This can cause that in a later step the view

![image](https://user-images.githubusercontent.com/87341702/222870046-5a4c6c77-df07-405c-9f4f-74ce2d8dfe51.png)


appears if the state of the bank account is PENDING because of this line:

https://github.com/Expensify/App/blob/0bd46f81f8f5dddc7c6dc8619cce05a94db85306/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L273

I'm not sure yet about what to do about this...

One option is to do something in [componentDidUpdate](https://github.com/Expensify/App/blob/3e15034b5e768393e56e7b6bf884bd2e3e61b135/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L99-L114) to recalculate this state.

Another option could be to move `shouldHideContinueSetupButton` to redux and put it inside the `reimbursementAccount` store... this last option can make it easy to update it with `failureData`/`successData` accordingly.

Reply:
I ended up refactoring this state. The bug shouldn't be there anymore.